src/d3d/nvapi_d3d_low_latency_device.cpp

```diff
@@ -1,6 +1,46 @@
 #include "nvapi_d3d_low_latency_device.h"
 
 namespace dxvk {
+    LowLatencyFrameIdGenerator::LowLatencyFrameIdGenerator():
+        m_currentLowLatencyDeviceFrameId(0)
+    {}
+
+    LowLatencyFrameIdGenerator::~LowLatencyFrameIdGenerator()
+    {}
+
+    uint64_t LowLatencyFrameIdGenerator::GetLowLatencyDeviceFrameId(uint64_t applicationFrameId) {
+        std::scoped_lock lock(m_frameIdGeneratorMutex);
+
+        auto it = m_applicationIdToDeviceId.find(applicationFrameId);
+        if (it != m_applicationIdToDeviceId.end())
+            return it->second;
+
+        uint64_t lowLatencyDeviceFrameId = m_currentLowLatencyDeviceFrameId++;
```

Comment:
Since we are incrementing the counter after retrieval, wouldn't a name like `m_nextLowLatencyDeviceFrameId` be more appropriate?

Or, alternatively we could do preincrementation here and keep the name, which has the added minor benefit of not having to deal with `0` being a valid device frame ID (because it already can't be a valid present ID).

Reply:
Would be good to not use 0 as a marker ID. It should be monotonically increasing over the lifetime of a device (not individual swapchain).